src/librustc/infer/error_reporting/mod.rs

```diff
@@ -1143,27 +1146,6 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                 }
                 (_, false, _) => {
                     if let Some(exp_found) = exp_found {
-                        let (def_id, ret_ty) = match exp_found.found.sty {
```

Comment:
Can you please explain this diff?

Reply:
We used to suggest `did you mean Foo(...)?`. Now we catch this case in the new code and I have a new commit that will be even clearer. If I hadn't removed this code we would be suggesting the same thing twice.